src/plugins/intel_cpu/thirdparty/ACLConfig.cmake

```
@@ -84,311 +103,376 @@ elseif(ENABLE_ARM_COMPUTE_CMAKE)
+                message(STATUS "SCONS: using ANDROID_TOOLCHAIN_PREFIX=${ANDROID_TOOLCHAIN_PREFIX_FIXED}")
+                ov_arm_compute_add_option("toolchain_prefix" "${ANDROID_TOOLCHAIN_PREFIX_FIXED}")
+            endif()
+
+            ov_arm_compute_add_option("compiler_prefix" "${ANDROID_TOOLCHAIN_ROOT}/bin/")
+
+            set(extra_cc_flags "--target=${ANDROID_LLVM_TRIPLE}" PARENT_SCOPE)
+            set(extra_flags "${extra_flags} --gcc-toolchain=${ANDROID_TOOLCHAIN_ROOT}" PARENT_SCOPE)
+            set(extra_flags "${extra_flags} --sysroot=${CMAKE_SYSROOT}" PARENT_SCOPE)
+
+        # Linux configuration
+        elseif(LINUX)
+            get_filename_component(cxx_compiler "${CMAKE_CXX_COMPILER}" NAME)
+            get_filename_component(c_compiler "${CMAKE_C_COMPILER}" NAME)
+            get_filename_component(compiler_prefix "${CMAKE_CXX_COMPILER}" DIRECTORY)
+
+            set(cmake_build_env
+                CC=${c_compiler}
+                CXX=${cxx_compiler} PARENT_SCOPE)
+            ov_arm_compute_add_option("compiler_prefix" "${compiler_prefix}/")
+
+        # Emscripten configuration
+        elseif(EMSCRIPTEN)
+            set(cmake_build_env
+                CC=emcc
+                CXX=em++
+                RANLIB=emranlib
+                AR=emar PARENT_SCOPE)
+
+            get_filename_component(toolchain_prefix "${CMAKE_CXX_COMPILER}" DIRECTORY)
+            ov_arm_compute_add_option("toolchain_prefix" "${toolchain_prefix}/")
+
+        # Apple/macOS configuration
+        elseif(APPLE)
+            get_filename_component(cxx_compiler "${CMAKE_CXX_COMPILER}" NAME)
+            get_filename_component(c_compiler "${CMAKE_C_COMPILER}" NAME)
+            get_filename_component(compiler_prefix "${CMAKE_CXX_COMPILER}" DIRECTORY)
+
+            set(cmake_build_env
+                CC=${c_compiler}
+                CXX=${cxx_compiler} PARENT_SCOPE)
+
+            ov_arm_compute_add_option("compiler_prefix" "${compiler_prefix}/")
+        endif()
+
+        set(ARM_COMPUTE_OPTIONS "${ARM_COMPUTE_OPTIONS}" PARENT_SCOPE)
+    endfunction()
+
+    # Helper function to configure compiler flags
+    function(ov_arm_compute_configure_flags)
+        set(local_extra_cxx_flags "${extra_cxx_flags}")
+        set(local_extra_link_flags "${extra_link_flags}")
+        set(local_extra_cc_flags "${extra_cc_flags}")
+
+        # Platform-specific flags
+        if(ANDROID)
+            set(extra_flags "${extra_flags} --gcc-toolchain=${ANDROID_TOOLCHAIN_ROOT}")
+            set(extra_flags "${extra_flags} --sysroot=${CMAKE_SYSROOT}")
+            set(local_extra_link_flags "${local_extra_link_flags} ${extra_flags}")
+            set(local_extra_cxx_flags "${local_extra_cxx_flags} ${extra_flags}")
+
+        elseif(EMSCRIPTEN)
+            set(local_extra_cxx_flags "${local_extra_cxx_flags} -msimd128")
+            set(local_extra_cxx_flags "${local_extra_cxx_flags} \
```

Please, do not remove related comments:

```suggestion
            # EMSDK: Passing any of -msse, -msse2, -msse3, -mssse3, -msse4.1, -msse4.2,
            # -msse4, -mavx, -mfpu=neon flags also requires passing -msimd128 (or -mrelaxed-simd)!
            set(local_extra_cxx_flags "${local_extra_cxx_flags} -msimd128")
            # clang-16: error: argument unused during compilation: '-mthumb' [-Werror,-Wunused-command-line-argument]
            # clang-16: error: argument unused during compilation: '-mfloat-abi=hard' [-Werror,-Wunused-command-line-argument]
            set(local_extra_cxx_flags "${local_extra_cxx_flags} \
```

Same for other encounters